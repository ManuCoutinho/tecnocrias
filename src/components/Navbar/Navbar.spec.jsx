import React from 'react'
import '@testing-library/jest-dom'
import { screen, render, fireEvent } from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import { describe, it, expect, vi } from 'vitest'
import { Navbar } from '.'

vi.spyOn(React, 'useState')
describe('<Navbar/>', () => {
  it('should render Navbar correctly ', () => {
    render(<Navbar/>)
    const home = screen.getByRole('menuitem', { name: /home/i })
    expect(screen.getByRole('navigation')).toBeInTheDocument()
    expect(screen.getByRole('menu')).toBeInTheDocument()
    expect(home).toBeInTheDocument()
    expect(home.parentNode).toHaveAttribute('href', '#home')
    expect(screen.getByRole('menuitem', { name: /sobre/i })).toBeInTheDocument()
    expect(
       screen.getByRole('menuitem', { name: /desafios/i })
     ).toBeInTheDocument()
    expect(
        screen.getByRole('menuitem', { name: /contato/i })
      ).toBeInTheDocument()
    expect(
         screen.getByRole('menuitem', { name: /comunidade/i })
       ).toBeInTheDocument()

  })
   it('should render the menu item with status active when selected', () => {
    render(<Navbar />)
    const about = screen.getByRole('link', { name: /sobre/i })
    expect(about).toHaveAttribute('data-active', 'false')
    act(() => {
      fireEvent.click(about)
    })
    expect(about).toHaveAttribute('data-active', 'true')
   })
  it('should match to snapshot', () => {
    const { container } = render(<Navbar/>)
    expect(container).toMatchSnapshot()
  });
 })
