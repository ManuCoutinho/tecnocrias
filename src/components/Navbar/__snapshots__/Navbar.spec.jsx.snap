// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Navbar/> > should match to snapshot 1`] = `
<div>
  <nav
    class="nav"
  >
    <ul
      aria-controls="menuitem"
      class="nav__menu"
      role="menu"
    >
      <a
        class="nav__item"
        data-active="false"
        href="#home"
      >
        <li
          role="menuitem"
          tabindex="1"
        >
          Home
        </li>
      </a>
      <a
        class="nav__item"
        data-active="false"
        href="#about"
      >
        <li
          role="menuitem"
          tabindex="1"
        >
          Sobre
        </li>
      </a>
      <a
        class="nav__item"
        data-active="false"
        href="#community"
      >
        <li
          role="menuitem"
          tabindex="1"
        >
          Comunidade
        </li>
      </a>
      <a
        class="nav__item"
        data-active="false"
        href="#challengers"
      >
        <li
          role="menuitem"
          tabindex="1"
        >
          Desafios
        </li>
      </a>
      <a
        class="nav__item"
        data-active="false"
        href="#contact"
      >
        <li
          role="menuitem"
          tabindex="1"
        >
          Contato
        </li>
      </a>
    </ul>
  </nav>
</div>
`;
